lib/phoenix/router/helpers.ex

```diff
@@ -231,6 +240,59 @@ defmodule Phoenix.Router.Helpers do
     end
   end
 
+  def defhelper_catch_all({helper, routes}, env) do
+    valid_routes = Enum.map(routes, &(&1.opts))
+    router = env.module
+    # We are using -1 to avoid warnings in case a path has already been defined.
+
+    for suffix <- ["path", "url"] do
+      quote line: -1 do
+        def unquote(:"#{helper}_#{suffix}")(_conn_or_endpoint, action) do
+          Phoenix.Router.Helpers.raise_route_error({__MODULE__, "#{unquote(helper)}_#{unquote(suffix)}", 2},
+                                                    unquote(router), unquote(helper), action, unquote(valid_routes))
+        end
+
+        def unquote(:"#{helper}_#{suffix}")(_conn_or_endpoint, action, _params) do
+          Phoenix.Router.Helpers.raise_route_error({__MODULE__, "#{unquote(helper)}_#{unquote(suffix)}", 3},
+                                                   unquote(router), unquote(helper), action, unquote(valid_routes))
+        end
+
+        def unquote(:"#{helper}_#{suffix}")(_conn_or_endpoint, action, action_2, _params) do
+          Phoenix.Router.Helpers.raise_route_error({__MODULE__, "#{unquote(helper)}_#{unquote(suffix)}", 4},
+                                                   unquote(router), unquote(helper), action, unquote(valid_routes))
+        end
+
+        def unquote(:"#{helper}_#{suffix}")(_conn_or_endpoint, action, action_2, action_3, _params) do
+          Phoenix.Router.Helpers.raise_route_error({__MODULE__, "#{unquote(helper)}_#{unquote(suffix)}", 5},
+                                                   unquote(router), unquote(helper), action, unquote(valid_routes))
+        end
+      end
+    end
+  end
+
+  @doc false
+  def raise_route_error({m, f, a}, router, helper, action, valid_routes) do
+    module_without_prefix = m |> to_string() |> String.split(".", parts: 2) |> Enum.at(1)
+    message = case action in valid_routes do
+      true ->
+        "No route helper clause for #{module_without_prefix}.#{f}/#{a} " <>
```

Comment:
What if we skip the module name? Most users won't care. Plus it will likely be in the stacktrace. Just keep fun + arity.